CRM/Extension/Upgrader/SchemaTrait.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * The SchemaTrait provides utilities for altering tables during an upgrade.
+ */
+trait CRM_Extension_Upgrader_SchemaTrait {
+
+  /**
+   * Add a column to a table if it doesn't already exist
+   *
+   * @param string $table
+   * @param string $column
+   * @param string $properties
+   *
+   * @return bool
+   */
+  public static function addColumn($table, $column, $properties) {
```

Comment:
Comparing these the `CRM_Upgrade_Incremental_Base` and `CRM_Extension_Upgrader`, they differ in their approaches to pagination/queueing (and specifically the `TaskContext $ctx` param).

* Core upgrades are universal (running on small+large DBs). There is value in paginating large data-updates. `CRM_Upgrade_Incremental_Base` makes the various `Queue`/`Task`/`TaskContext` objects more visible so that it's easier to paginate.
* Extensions aren't universal -- and large/paginated updates are not expected as frequently for them. So `CRM_Extension_Upgrader` presents a simpler/flatter facade.

If one really wants DRY, there's probably some clever refactoring... (Maybe a variation on `addTask()` could obscure `$ctx` and make it easier to use the same functions as `CRM_Extension_Upgrader`?)

But I also think it's fair to say: "`CRM_Extension_Upgrader` provides a typical/normal/simplified base-class for extensions -- but extensions are free to use different base-classes -- so there's no need for one class to be the end-all/be-all."

Of course, if extensions can multilingual schema, then it's good for `CRM_Extension_Upgrader` to support that...

Reply:
We could always add that support later.
Any objections to merging this?

Reply:
No objections from me wrt core-upgrader / extension-upgrader / alignment. (The split happened long ago - and faffing about one helper here won't change that calculus.)

But I do think it would be good to get feedback on the multilingual question (eg ping @samuelsov or @bgm) -- this function would become a public API, and it's better to wait a day for extra feedback than to have the semantics bounce-around.

Another observation: if the multilingual impl  in `CRM_Upgrade_Incremental_Base::addColumn` is suitable to ext's, you could probably wrap it.

```php
public function addColumn($table, $column, $properties): bool {
  return CRM_Upgrade_Incremental_Base::addColumn(NULL, $table, $column, $properties);
}
```

Reply:
Since the other 3 functions are ever-so-close-to-identical to the same functions in Incremental::Base, why not either make all the trait functions wrappers, or convert Incremental::Base to share the trait.

Reply:
We could do that refactor at some point, just don't have time for it right now.